precompile/contracts/hubblebibliophile/contract.go

```diff
@@ -159,14 +190,112 @@ func getPositionSizes(accessibleState contract.AccessibleState, caller common.Ad
+	inputStruct, err := UnpackValidateLiquidationOrderAndDetermineFillPriceInput(input)
+	if err != nil {
+		return nil, remainingGas, err
+	}
+
+	// CUSTOM CODE STARTS HERE
+	output, err := ValidateLiquidationOrderAndDetermineFillPrice(accessibleState.GetStateDB(), &inputStruct)
+	if err != nil {
+		return nil, remainingGas, err
+	}
+	packedOutput, err := PackValidateLiquidationOrderAndDetermineFillPriceOutput(output)
+	if err != nil {
+		return nil, remainingGas, err
+	}
+
+	// Return the packed output and the remaining gas
+	return packedOutput, remainingGas, nil
+}
+
+// UnpackValidateOrdersAndDetermineFillPriceInput attempts to unpack [input] as ValidateOrdersAndDetermineFillPriceInput
+// assumes that [input] does not include selector (omits first 4 func signature bytes)
+func UnpackValidateOrdersAndDetermineFillPriceInput(input []byte) (ValidateOrdersAndDetermineFillPriceInput, error) {
+	inputStruct := ValidateOrdersAndDetermineFillPriceInput{}
+	err := HubbleBibliophileABI.UnpackInputIntoInterface(&inputStruct, "validateOrdersAndDetermineFillPrice", input)
+
+	return inputStruct, err
+}
+
+// PackValidateOrdersAndDetermineFillPrice packs [inputStruct] of type ValidateOrdersAndDetermineFillPriceInput into the appropriate arguments for validateOrdersAndDetermineFillPrice.
+func PackValidateOrdersAndDetermineFillPrice(inputStruct ValidateOrdersAndDetermineFillPriceInput) ([]byte, error) {
+	return HubbleBibliophileABI.Pack("validateOrdersAndDetermineFillPrice", inputStruct.Orders, inputStruct.OrderHashes, inputStruct.FillAmount)
+}
+
+// PackValidateOrdersAndDetermineFillPriceOutput attempts to pack given [outputStruct] of type ValidateOrdersAndDetermineFillPriceOutput
+// to conform the ABI outputs.
+func PackValidateOrdersAndDetermineFillPriceOutput(outputStruct ValidateOrdersAndDetermineFillPriceOutput) ([]byte, error) {
+	return HubbleBibliophileABI.PackOutput("validateOrdersAndDetermineFillPrice",
+		outputStruct.FillPrice,
+		outputStruct.Mode0,
+		outputStruct.Mode1,
+	)
+}
+
+func validateOrdersAndDetermineFillPrice(accessibleState contract.AccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
+	if remainingGas, err = contract.DeductGas(suppliedGas, ValidateOrdersAndDetermineFillPriceGasCost); err != nil {
+		return nil, 0, err
+	}
+	// attempts to unpack [input] into the arguments to the ValidateOrdersAndDetermineFillPriceInput.
+	// Assumes that [input] does not include selector
+	// You can use unpacked [inputStruct] variable in your code
+	inputStruct, err := UnpackValidateOrdersAndDetermineFillPriceInput(input)
+	if err != nil {
+		return nil, remainingGas, err
+	}
+
+	// CUSTOM CODE STARTS HERE
+	output, err := ValidateOrdersAndDetermineFillPrice(accessibleState.GetStateDB(), &inputStruct)
+	if err != nil {
+		return nil, remainingGas, err
+	}
+	packedOutput, err := PackValidateOrdersAndDetermineFillPriceOutput(*output)
```

Comment:
ValidateOrdersAndDetermineFillPrice is returning nil in some cases. Here `*output` in that case will give  nil pointer dereference error. 
Either pass `PackValidateOrdersAndDetermineFillPriceOutput` pointer itself and then abi pack will take care of packing default values
or return pointer of empty `ValidateOrdersAndDetermineFillPriceOutput` in ValidateOrdersAndDetermineFillPrice function